### ISSUE TITLE
Bug: Urls with hash symbol in them do not get "angle bracketed" correctly

### ISSUE THREAD

**Issue by finn-matti** [platers/obsidian-linter#959]
- [x] I have verified that I am on the latest version of the Linter

## Describe the Bug

If you have a link like "https://github.com/platers/obsidian-linter//issues/42#issuecomment-1234567890" and "Encloses bare URLs with angle brackets except when enclosed in back ticks, square braces, or single or double quotes." activated, on the latest version, only the part in front of the hash sign gets put into the brackets.

## How to Reproduce

1. Activate the rule
2. Copy the example above and paste into obsidian
3. Lint

```markdown
<https://github.com/platers/obsidian-linter//issues/42>#issuecomment-1234567890
```

## Expected Behavior

``` markdown
<https://github.com/platers/obsidian-linter//issues/42#issuecomment-1234567890>
```

## Screenshots

![Screenshot 2023-12-04 at 09 45 34](https://github.com/platers/obsidian-linter/assets/56148301/c20313b4-93e9-488b-9fae-f186899f9453)

## Device

- [x] Desktop
- [ ] Mobile - didn't test

**Comment by pjkaufman** [comment]
Yeah. It looks like the issue is that the hash symbol is not properly recognized. A one line change seems to fix it. I will get the change merged in and it should go out with the next release.

**Comment by finn-matti** [comment]
Nice. Thanks for the quick response and resolution. 🥳
